odo should exit with error when s2i build fails

```
$ odo create wildfly backend --binary target/ROOT.war
Receiving source from STDIN as file ROOT.war
pulling image error : pulling image "docker.io/openshift/wildfly-101-centos7@sha256:a2661f20f2e870520a73765f783de105c61916235a541332acd1501c2ee76280" timed out after 2m0s
error: build error: unable to get docker.io/openshift/wildfly-101-centos7@sha256:a2661f20f2e870520a73765f783de105c61916235a541332acd1501c2ee76280
Component 'backend' was created.

Component 'backend' is now set as active component.
```

There should be a message informing the user that component creation failed.
The same for push.
```
$ odo push
Pushing changes to component: backend
Receiving source from STDIN as file ROOT.war
pulling image error : pulling image "docker.io/openshift/wildfly-101-centos7@sha256:a2661f20f2e870520a73765f783de105c61916235a541332acd1501c2ee76280" timed out after 2m0s
error: build error: unable to get docker.io/openshift/wildfly-101-centos7@sha256:a2661f20f2e870520a73765f783de105c61916235a541332acd1501c2ee76280
changes successfully pushed to component: backend
```

This issue has two parts.

The first part is a message that is displayed when `odo create` fails.
After #408 has been merged It behaves a bit differently. `odo create` now creates "empty" component by injecting Supervisor into builder image using BuildConfig (s2i).
If there is an error during that process `odo` exits with exit code 1. That is correct behaviour. 
But the message that is displayed is not user-friendly and might be confusing for end user.
```
▶ odo create wildfly
Please wait, creating wildfly component ...
build wildfly-1 status Failed

▶ echo $?
1
```
Instead of `build wildfly-1 status Failed` we should display some nice message like. "Error while preparing component" or something like that. 



The second part is detecting errors while running `odo push`.
`odo push` upload the source code or the binary to the container, than it runs [assemble-and-restart.sh](https://github.com/kadel/bootstrap-supervisored-s2i/blob/master/scripts/assemble-and-restart.sh) script. This scripts calls s2i assemble script and than reloads supervisor service.
Problem is when s2i assemble script fails (for example due too syntax error). It would be nice if we could detect that and make `odo push` return error exit code 
```
▶ odo push
Pushing changes to component: wildfly
building file list ... done
target/classes/eu/mjelen/katacoda/odo/BackendServlet.class
target/maven-status/maven-compiler-plugin/compile/default-compile/createdFiles.lst
target/maven-status/maven-compiler-plugin/compile/default-compile/inputFiles.lst

sent 3077 bytes  received 92 bytes  6338.00 bytes/sec
total size is 16397  speedup is 5.17
Please wait, building component....
touch: cannot touch '/tmp/src/.dummy': No such file or directory
Found pom.xml... attempting to build with 'mvn package -Popenshift -DskipTests -B -s /opt/app-root/src/.m2/settings.xml'
Apache Maven 3.3.9 (bb52d8502b132ec0a5a3f4c09453c07478323dc5; 2015-11-10T16:41:47+00:00)
Maven home: /usr/local/apache-maven-3.3.9
Java version: 1.8.0_161, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.161-0.b14.el7_4.x86_64/jre
Default locale: en_US, platform encoding: ANSI_X3.4-1968
OS name: "linux", version: "4.4.41-boot2docker", arch: "amd64", family: "unix"
[INFO] Scanning for projects...
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building PersistentStorageCounterServlet 1
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ backend ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /opt/app-root/src/src/main/resources
[INFO]
[INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ backend ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 1 source file to /opt/app-root/src/target/classes
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] /opt/app-root/src/src/main/java/eu/mjelen/katacoda/odo/BackendServlet.java:[26,21] ';' expected
[INFO] 1 error
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 1.213 s
[INFO] Finished at: 2018-05-10T12:00:17+00:00
[INFO] Final Memory: 10M/95M
[INFO] ------------------------------------------------------------------------
[WARNING] The requested profile "openshift" could not be activated because it does not exist.
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project backend: Compilation failure
[ERROR] /opt/app-root/src/src/main/java/eu/mjelen/katacoda/odo/BackendServlet.java:[26,21] ';' expected
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
Aborting due to error code 1 from mvn package
run: stopped
run: started
changes successfully pushed to component: wildfly

▶ echo $?
0
```
Also message "changes successfully pushed to component: wildfly" might be a bit confusing in this case.

--- a/pkg/occlient/occlient.go
+++ b/pkg/occlient/occlient.go
@@ -651,7 +651,7 @@ func (c *Client) NewAppS2I(name string, builderImage string, gitUrl string, labe
 func (c *Client) BootstrapSupervisoredS2I(name string, builderImage string, labels map[string]string, annotations map[string]string) error {
 	// git repository that will be used for bootstraping
 	const bootstrapperURI = "https://github.com/kadel/bootstrap-supervisored-s2i"
-	const bootstrapperRef = "v0.0.1"
+	const bootstrapperRef = "v0.0.2"
 
 	appRootVolumeName := fmt.Sprintf("%s-s2idata", name)
 

